client/forest-manager/src/rocksdb.rs

```diff
@@ -0,0 +1,611 @@
+use std::{io, path::PathBuf, sync::Arc};
+
+use hash_db::{AsHashDB, HashDB, Prefix};
+use kvdb::{DBTransaction, KeyValueDB};
+use kvdb_rocksdb::{Database, DatabaseConfig};
+use log::debug;
+use sp_state_machine::{warn, Storage};
+use sp_trie::{
+    prefixed_key, recorder::Recorder, PrefixedMemoryDB, Trie, TrieDBBuilder, TrieLayout, TrieMut,
+};
+use storage_hub_infra::types::{ForestProof, Metadata};
+use trie_db::{DBValue, Hasher, TrieDBMutBuilder};
+
+use crate::{
+    prove::prove,
+    traits::ForestStorage,
+    types::{ForestStorageErrors, HashT, HasherOutT, RawKey},
+    utils::{deserialize_value, serialize_value},
+};
+
+const LOG_TARGET: &str = "forest_storage";
+
+mod well_known_keys {
+    pub const ROOT: &[u8] = b":root";
+}
+
+pub(crate) fn other_io_error(err: String) -> io::Error {
+    io::Error::new(io::ErrorKind::Other, err)
+}
+
+/// Open the database on disk, creating it if it doesn't exist.
+fn open_creating_rocksdb(db_path: String) -> io::Result<Database> {
+    let root = PathBuf::from("/tmp/");
+    let path = root.join("storagehub").join(db_path);
+
+    let db_config = DatabaseConfig::with_columns(1);
+
+    let path_str = path
+        .to_str()
+        .ok_or_else(|| other_io_error(format!("Bad database path: {:?}", path)))?;
+
+    std::fs::create_dir_all(&path_str)?;
+    let db = Database::open(&db_config, &path_str)?;
+
+    Ok(db)
+}
+
+/// Storage backend for RocksDB.
+pub struct StorageDb<Hasher> {
+    pub db: Arc<dyn KeyValueDB>,
+    pub _phantom: std::marker::PhantomData<Hasher>,
+}
+
+impl<H: Hasher> Storage<H> for StorageDb<H> {
+    fn get(&self, key: &H::Out, prefix: Prefix) -> Result<Option<DBValue>, String> {
+        let prefixed_key = prefixed_key::<H>(key, prefix);
+        self.db.get(0, &prefixed_key).map_err(|e| {
+            warn!(target: LOG_TARGET, "Failed to read from DB: {}", e);
+            format!("Failed to read from DB: {}", e)
+        })
+    }
+}
+
+/// Backend storage used by the [`RocksDBForestStorage`].
```

Comment:
```suggestion
/// Trait that a storage backend need to implement to be used by the [`RocksDBForestStorage`].
```

Reply:
Otherwise it is confusing with the `StorageDb` struct, which is commented to be the "Storage backend for RocksDB".